data_model/src/expression.rs

```diff
@@ -338,962 +499,561 @@ impl ContextValue {
+}
+
+gen_expr_and_impls! {
+    /// `ContainsAny` expression.
+    /// Returns `true` if `collection` contains any element out of the `elements`, `false` otherwise.
+    #[derive(
+        Debug,
+        Display,
+        Clone,
+        PartialEq,
+        Eq,
+        Decode,
+        Encode,
+        Deserialize,
+        Serialize,
+        IntoSchema,
+        PartialOrd,
+        Ord,
+    )]
+    #[display(
+        fmt = "{}.contains_any({})",
+        "collection.parenthesise(Operation::MethodCall)",
+        "elements"
+    )]
+    pub ContainsAny(collection: Vec<Value>, elements: Vec<Value>) -> bool
+}
+
+gen_expr_and_impls! {
+    /// Returns `true` if `left` operand is equal to the `right` operand.
+    #[derive(
+        Debug,
+        Display,
+        Clone,
+        PartialEq,
+        Eq,
+        Decode,
+        Encode,
+        Deserialize,
+        Serialize,
+        IntoSchema,
+        PartialOrd,
+        Ord,
+    )]
+    #[display(
+        fmt = "{} == {}",
+        "self.left.parenthesise(Operation::Equal)",
+        "self.right.parenthesise(Operation::Equal)"
+    )]
+    pub Equal(Value, Value) -> bool
 }
 
-impl Mod {
-    /// Number of underneath expressions.
-    pub fn len(&self) -> usize {
-        self.left.len() + self.right.len() + 1
-    }
+/// [`Where`] builder.
+#[derive(Debug)]
+pub struct WhereBuilder {
+    /// Expression to be evaluated.
+    expression: EvaluatesTo<Value>,
+    /// Context values for the context binded to their `String` names.
+    values: btree_map::BTreeMap<ValueName, EvaluatesTo<Value>>,
+}
 
-    /// Constructs `Mod` expression.
-    pub fn new(left: impl Into<EvaluatesTo<u32>>, right: impl Into<EvaluatesTo<u32>>) -> Self {
+impl WhereBuilder {
+    /// Sets the `expression` to be evaluated.
+    #[must_use]
+    pub fn evaluate<E: Into<EvaluatesTo<Value>>>(expression: E) -> Self {
         Self {
-            left: left.into(),
-            right: right.into(),
+            expression: expression.into(),
+            values: btree_map::BTreeMap::new(),
         }
     }
-}
 
-impl From<Mod> for ExpressionBox {
-    fn from(expression: Mod) -> Self {
-        Expression::Mod(expression).into()
+    /// Binds `expression` result to a `value_name`, by which it will be reachable from the main expression.
+    #[must_use]
+    pub fn with_value<E: Into<EvaluatesTo<Value>>>(
+        mut self,
+        value_name: ValueName,
+        expression: E,
+    ) -> Self {
+        let _result = self.values.insert(value_name, expression.into());
+        self
+    }
+
+    /// Returns a [`Where`] expression.
+    #[inline]
+    #[must_use]
+    pub fn build(self) -> Where {
+        Where::new(self.expression, self.values)
     }
 }
 
-/// Evaluates to the right expression in power of left expressions.
-/// Works only for `Value::U32`
+/// Adds a local context of `values` for the `expression`.
+/// It is similar to *Haskell's where syntax* although, evaluated eagerly.
```

Comment:
```suggestion
/// It is similar to **where** syntax in Haskell.
```

Not sure what you want to say with `evaluated eagerly`

Reply:
I think, this comment was first introduced by @appetrosyan. It's a terminology in computer science that applies to the expression evaluation order

Reply:
Actually It was introduced by @i1i1r, but the gist is that the `where` block is evaluated as soon as possible, and not when it's actually used.